dashboard/app/models/user.rb

```diff
@@ -151,6 +153,23 @@ class User < ActiveRecord::Base
   has_many :districts_users, class_name: 'DistrictsUsers'
   has_many :districts, through: :districts_users
 
+  belongs_to :school_info
+  accepts_nested_attributes_for :school_info, reject_if: :check_school_info
+  validates_presence_of :school_info, unless: :school_info_optional?
+
+  def check_school_info(school_info_attr)
+    # Suppress validation - all parts of this form are optional when accesssed through the registration form
+    school_info_attr[:validation_type] = SchoolInfo::VALIDATION_NONE unless school_info_attr.nil?
+    # students are never asked to fill out this data, so silently reject it for them
+    student? || deduplicate_school_info(school_info_attr, self)
+  end
+
+  # Not deployed to everyone, so we don't require this for anybody, yet
+  def school_info_optional?
+    true
+    #student? || (persisted? && created_at < '2017-MM-DD') # update if/when A/B test is done and accepted
```

Comment:
That's copied from enrollment.rb, I wasn't entirely sure what the goal was with that condition. For now I'll probably delete that, if and when we want to make this mandatory we can figure out the best way to filter.